network/transport/types.go

```diff
@@ -33,12 +35,61 @@ func (p PeerID) String() string {
 	return string(p)
 }
 
+// Addr describes an address of a node on the Nuts Network.
+type Addr struct {
+	scheme string
+	target string
+}
+
+// Empty returns whether the Addr is considered empty.
+func (addr Addr) Empty() bool {
+	return len(addr.scheme) == 0 || len(addr.target) == 0
+}
+
+// Scheme returns the protocol scheme of the address (e.g. "grpc").
+func (addr Addr) Scheme() string {
+	return addr.scheme
+}
+
+// Target returns the target of the address (e.g. "10.0.0.1:5555").
+func (addr Addr) Target() string {
+	return addr.target
+}
+
+// String returns the fully qualifies address, which includes the protocol scheme (e.g. "grpc").
+func (addr Addr) String() string {
+	if addr.Empty() {
+		return ""
+	}
+	return addr.scheme + "://" + addr.target
+}
+
+// Address makes a new gRPC Addr given the target address.
+func Address(target string) Addr {
+	return Addr{
+		scheme: "grpc", // we only support gRPC right now
+		target: target,
+	}
+}
+
+// ParseAddress parses the given input string to an Addr. The input must include the protocol scheme (e.g. grpc://).
+func ParseAddress(input string) (Addr, error) {
+	parsed, err := url.Parse(input)
+	if err != nil {
+		return Addr{}, err
+	}
+	if parsed.Scheme != "grpc" {
+		return Addr{}, errors.New("invalid URL scheme")
+	}
+	return Addr{scheme: parsed.Scheme, target: parsed.Host}, nil
+}
```

Comment:
Why not use `url.URL` type instead of creating an `Addr` type?

Reply:
I initially had it as `url.URL` but then you can supply many variants which don't qualify as a valid gRPC target address. We currently only support the DNS-style addresses, but there are a few other styles I can imagine we want to support (e.g. for clustering and/or in IPv6 environments), see https://github.com/grpc/grpc/blob/master/doc/naming.md

Therefore I figured the safest and most future-proof way to implement this was to introduce a type which implementation can be extended to support new/other formats.

Reply:
This is the actual target parsing code in the `grpc-go` library: https://github.com/grpc/grpc-go/blob/14c11384b76b67f7b1b32a5d18f865762634c0ae/clientconn.go#L1663 Looking at it this seems wrong. To be honest I'm not even sure why we need an `Addr` type at all. What's wrong with using a `string` here?

Reply:
If all that is needed is supported by the grpc library itself, then use that?

Reply:
Reverted it back to `string`. We can't use the internal type since it's private, unfortunately.